G1ANT.Addon.Net/Structures/AttachmentStructure.cs

```diff
@@ -0,0 +1,122 @@
+            : base(value, format, scripter)
+        {
+            Init();
+        }
+
+        private void Init(ITempFileService iTempFileService = null, IGetMd5HashService iGetMd5HashService = null)
+        {
+            Indexes.Add(NameIndex);
+            Indexes.Add(SizeIndex);
+            Indexes.Add(TypeIndex);
+            Indexes.Add(PathIndex);
+            tempFileService = iTempFileService ?? new TempFileService();
+            getMd5HashService = iGetMd5HashService ?? new GetMd5HashService();
+        }
+
+        public override Structure Get(string index = "")
+        {
+            if (string.IsNullOrWhiteSpace(index))
+            {
+                throw new NotSupportedException("Index not found in AttachmentStructure");
+            }
+
+            switch (index.ToLower())
+            {
+                case NameIndex:
+                    return new TextStructure(Value.Name, null, Scripter);
+                case SizeIndex:
+                    return new IntegerStructure(Value.Size, null, Scripter);
+                case TypeIndex:
+                    return new TextStructure(Value.Type, null, Scripter);
+                case PathIndex:
+                    pathToTempFile = SaveAndGetPath(Value.MimeEntity, Value.Name);
+                    return new PathStructure(pathToTempFile, null, Scripter);
+                default:
+                    throw new ArgumentException($"Unknown index '{index}'", nameof(index));
+            }
+
+        }
+
+        private string SaveAndGetPath(MimeEntity mimeEntity, string fileName)
+        {
+            var attachmentNameHash = GetAttachmentNameHash(fileName, mimeEntity.ContentId);
+            var filePath = GetAttachmentTempFileNamePath($"{attachmentNameHash}.{fileName}");
+            if (!File.Exists(filePath))
+            {
+                if (mimeEntity is MessagePart msgPart)
+                {
+                    using (var stream = File.Create(filePath))
+                        msgPart.Message.WriteTo(stream);
+                }
+                else if (mimeEntity is MimePart mimePart)
+                {
+                    using (var stream = File.Create(filePath))
+                        mimePart.Content.DecodeTo(stream);
+                }
+                else
+                    return string.Empty;
+            }
+            return filePath;
+        }
+
+        public override string ToString(string format = "")
+        {
+            return Value.Name;
+        }
+
+        private string GetAttachmentTempFileNamePath(string attachmentNameHash)
+        {
+            return tempFileService.GetTempPath(AttachmentFilePrefix, attachmentNameHash, "");
+        }
+
+        private string GetAttachmentNameHash(string fileName, string contentId)
+        {
+            return getMd5HashService.GetMd5Hash($"{contentId}-{fileName}");
+        }
+
+        private void DisposeAttachment(string attachmentPath)
```

Comment:
DisposeAttachment -> RemoveAttachmentFile

Reply:
This code has been moved from G1ANT.Language, didn't you notice that? It is the name from original class